### ISSUE TITLE
TileLayer Locking

### ISSUE THREAD

**Issue by ilan-gold** [visgl/deck.gl#5041]
#### Description
It seems that if you zoom/pan/interact with a `TileLayer` application for long enough, it locks.  I haven't been able to fully determine why this is happening - I think I notice that the `requestScheduler` eventually stops giving "`priority`" for new tiles to fetch, but maybe I was mis-reading that.

#### Repro Steps
Zoom/pan somewhat haphazardly/quickly for a minute or two on a `TileLayer` (which I recognize may be an abuse of the server, but I also have seen this happen much quicker than a minute or two).  Maybe this is just an extreme edge-case, but I wanted to bring it up to see if there is a condition or something in the code that might be a known weak-point.  I'm going to keep digging in the interim as well.  I also recognize these are not great "Repro steps" but if people don't think this is really an issue, I can also close this.

#### Environment (please complete the following information):
 - Framework Version: 8.3.0
 - Browser Version: 81.0.2 (64-bit)
 - OS: 10.14.6

#### Screen-gif
![ezgif com-video-to-gif(8)](https://user-images.githubusercontent.com/43999641/96352421-fe236680-1090-11eb-8706-71c62955d1a9.gif)

**Comment by Pessimistress** [comment]
I believe this was introduced by the abort controller PR:

https://github.com/visgl/deck.gl/blob/master/modules/geo-layers/src/tile-layer/tile-2d-header.js#L58-L61

If the tile is aborted, `requestToken` is never marked `done`. @kylebarron @supersonicclay

**Comment by ilan-gold** [comment]
Yes I was mucking around the `esm` build @Pessimistress and I noticed that newly created layers would not go past this point.  That certainly sounds correct.

**Comment by kylebarron** [comment]
I'm guessing the issue is from this abort https://github.com/visgl/deck.gl/pull/4838/files#diff-ebfc47135e3d8fe2bee8b3d33fae4ebbbc295d01887c751d4cb77f71fc85a87aR192, since the abort from the AbortController happens only after the fetch has already been initiated.

I guess you want:

```diff
- if (!requestToken || this._isCancelled) {
-   this._isCancelled = true;
-   return;
- }
+ if (this._isCancelled) {
+   requestToken && requestToken.done();
+   return;
+ }
+ if (!requestToken) {
+   this._isCancelled = true;
+   return;
+ }
```
?

**Comment by Pessimistress** [comment]
Published 8.3.2

**Comment by ilan-gold** [comment]
Thanks!
